ldap/servers/plugins/uiduniq/uid.c

```diff
@@ -788,12 +788,12 @@ searchAllSubtrees(Slapi_DN **subtrees, Slapi_DN **exclude_subtrees, const char *
            * are unique in all the monitored subtrees
            */
 
-        /* First check the target entry is in one of
+        /* First check the source entry is in one of
            * the monitored subtree, so adding 'values' would
            * violate constraint
            */
         for (i = 0; subtrees && subtrees[i]; i++) {
-            if (slapi_sdn_issuffix(dn, subtrees[i])) {
+            if (slapi_sdn_issuffix(sourceSDN, subtrees[i])) {
```

Comment:
Here we should rather check if the destinationSDN lands into one of the monitored subtrees.